TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.CfgSentence.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections.Generic;
+using TypeCobol.Compiler.Symbols;
+
+namespace TypeCobol.Analysis.Cfg
+{
+    public partial class ControlFlowGraphBuilder<D>
+    {
+        /// <summary>
+        /// A Sentence used for our builder. A Sentence is a special symbol.
+        /// </summary>
+        internal class CfgSentence : Symbol
+        {
+            /// <summary>
+            /// First Block associated to this sentence.
+            /// </summary>
+            internal BasicBlockForNode Block
+            {
+                get;
+                set;
+            }
+
+            /// <summary>
+            /// The Block Index Associated to the Block.
+            /// </summary>
+            internal int BlockIndex
+            {
+                get;
+                set;
+            }
+
+            /// <summary>
+            /// All blocks in this Sentence.
+            /// </summary>
+            internal LinkedList<BasicBlockForNode> AllBlocks
+            {
+                get;
+                set;
+            }
+
+            /// <summary>
+            /// Add a block in this sequence.
+            /// </summary>
+            /// <param name="block">The block to be added</param>
+            internal void AddBlock(BasicBlockForNode block)
+            {
+                if (AllBlocks == null)
+                {
+                    AllBlocks = new LinkedList<BasicBlockForNode>();
+                }
+                AllBlocks.AddLast(block);
+            }
+
+            /// <summary>
+            /// Sentence counter
+            /// </summary>
+            private static int SentenceCounter = 0;
```

Comment:
This won't work in multiple parsing scenarios unless the counter is reset at some point.

Reply:
No in fact no this just an id for the sentence used to give a distinct name to a sentence. it's a static field we don't care about concurrency here. an **int** has 214748364750 values, it can be changed to an **uint** to have 4294967295 values, don't worry.

Reply:
See PR #1738. As Sentence is not a Symbol anymore, this counter has been removed.